File: src/main/java/com/practice/survey/surveyItemOption/model/entity/SurveyItemOption.java
package com.practice.survey.surveyItemOption.model.entity;

import com.practice.survey.common.model.entity.BaseTime;
import com.practice.survey.surveyItem.model.entity.SurveyItem;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Entity
public class SurveyItemOption extends BaseTime {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long optionId;

    @Column(nullable = false)
    private int optionNumber;

    @Column(columnDefinition = "TEXT",nullable = false)
    private String optionText;

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "item_id", nullable = false)
    private SurveyItem item;
}

File: survey-core/src/main/kotlin/org/icd/surveycore/domain/surveyItem/SurveyItemOption.kt
package org.icd.surveycore.domain.surveyItem

import jakarta.persistence.*
import org.hibernate.annotations.Comment

@Entity
@Table(name = "survey_item_option")
class SurveyItemOption(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "survey_item_id")
    val surveyItem: SurveyItem,
    @Comment("선택 후보 이름")
    var name: String,
    @Comment("현재 사용 여부")
    var isActive: Boolean = true
) {
    companion object {
        fun of(
            surveyItem: SurveyItem,
            name: String
        ): SurveyItemOption {
            return SurveyItemOption(
                surveyItem = surveyItem,
                name = name
            )
        }
    }

    fun delete() {
        this.isActive = false
    }

    fun updateName(name: String) {
        this.name = name
    }
}